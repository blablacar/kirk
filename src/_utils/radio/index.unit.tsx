import React from 'react'
import renderer from 'react-test-renderer'

import Radio from './index'

describe('Radio', () => {
  it('Should render an empty circle by default', () => {
    const radio = renderer.create(<Radio />).toJSON()
    expect(radio).toMatchSnapshot()
  })
  it('Should render a check icon if isChecked is set to true', () => {
    const radio = renderer.create(<Radio isChecked />).toJSON()
    expect(radio).toMatchSnapshot()
  })
  it('Should render a loader if isLoading is set to true', () => {
    const radio = renderer.create(<Radio isLoading />).toJSON()
    expect(radio).toMatchSnapshot()
  })
  it('Should render a loader if isLoading and isChecked are set to true', () => {
    const radio = renderer.create(<Radio isChecked isLoading />).toJSON()
    expect(radio).toMatchSnapshot()
  })
})
