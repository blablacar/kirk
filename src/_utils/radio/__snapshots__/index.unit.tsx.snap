// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Should render a check icon if isChecked is set to true 1`] = `
<svg
  aria-hidden={true}
  className="jsx-1131348929 kirk-icon"
  height={24}
  viewBox="0 0 66 66"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  
  <circle
    className="jsx-1131348929"
    cx="33"
    cy="33"
    fill="none"
    r="30"
    stroke="#00AFF5"
  />
  <circle
    className="jsx-1131348929 inner"
    cx="33"
    cy="33"
    fill="#00AFF5"
    r="18"
  />
</svg>
`;

exports[`Checkbox Should render a loader if isLoading and isChecked are set to true 1`] = `
<div
  className="jsx-2765694959 kirk-loader"
  style={
    Object {
      "height": "24px",
      "width": "24px",
    }
  }
>
  <svg
    aria-hidden={true}
    className="jsx-1131348929 kirk-icon spinning"
    height={24}
    viewBox="0 0 66 66"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    
    <circle
      className="jsx-1131348929"
      cx="33"
      cy="33"
      fill="none"
      r="30"
      stroke="#5DD167"
    />
  </svg>
</div>
`;

exports[`Checkbox Should render a loader if isLoading is set to true 1`] = `
<div
  className="jsx-2765694959 kirk-loader"
  style={
    Object {
      "height": "24px",
      "width": "24px",
    }
  }
>
  <svg
    aria-hidden={true}
    className="jsx-1131348929 kirk-icon spinning"
    height={24}
    viewBox="0 0 66 66"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    
    <circle
      className="jsx-1131348929"
      cx="33"
      cy="33"
      fill="none"
      r="30"
      stroke="#5DD167"
    />
  </svg>
</div>
`;

exports[`Checkbox Should render an empty circle by default 1`] = `
<svg
  aria-hidden={true}
  className="jsx-1131348929 kirk-icon"
  height={24}
  viewBox="0 0 66 66"
  width={24}
  xmlns="http://www.w3.org/2000/svg"
>
  
  <circle
    className="jsx-1131348929"
    cx="33"
    cy="33"
    fill="none"
    r="30"
    stroke="#00AFF5"
  />
</svg>
`;
